src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs

```diff
@@ -181,43 +181,49 @@ private static (Dictionary<StringSpanOrdinalKey, int>?, Dictionary<int, string>?
             {
                 using (StreamReader reader = new StreamReader(vocabStream))
                 {
-                    while (true)
+                    string? line;
+                    do
                     {
-                        string? line = useAsync ?
+                        line = useAsync ?
                             await Helpers.ReadLineAsync(reader, cancellationToken).ConfigureAwait(false) :
                             reader.ReadLine();
-                        if (string.IsNullOrWhiteSpace(line))
+                    } while (line is not null && line.Length == 0);
+
+                    if (line is not null && line.IndexOf(' ') < 0)
+                    {
+                        // We generate the ranking using the line number
```

Comment:
Should we add tests that can validate we produce the same vocab with our optimized format?

Reply:
I added the test to do that.